drivers/mtd/mtd.c

```diff
@@ -168,10 +167,47 @@ int mtd_write_page(mtd_dev_t *mtd, const void *data, uint32_t page,
     }
 
     /* modify sector in RAM */
-    memcpy(work + (page - sector_page) * mtd->page_size + offset, data, len);
+    memcpy(work + offset, data, len);
 
     /* write back modified sector copy */
-    return mtd_write_page_raw(mtd, work, sector_page, 0, sector_size);
+    res = mtd_write_page_raw(mtd, work, sector_page, 0, sector_size);
+    if (res < 0) {
+        return res;
+    }
+
+    return len;
+}
+
+int mtd_write_page(mtd_dev_t *mtd, const void *data, uint32_t page,
+                   uint32_t offset, uint32_t len)
+{
+    if (!mtd || !mtd->driver) {
+        return -ENODEV;
+    }
+
+    if (mtd->work_area == NULL) {
+        return mtd_write_page_raw(mtd, data, page, offset, len);
+    }
+
+    uint32_t sector = page / mtd->pages_per_sector;
+    const uint32_t sector_page = sector * mtd->pages_per_sector;
+    const char *_src = data;
```

Comment:
Just nitpicking but why the underscor?

Reply:
I did that elsewhere when the function itself have a `src` parameter, but that's not the case here - can remove it.